### PR TITLE
#234: converting all oracle timestamp into timestamp format in Exasol

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**

### CODE REVIEW

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleColumnMetadataReader.java [hunk 1]
```diff
@@ -44,13 +44,15 @@ public DataType mapJdbcType(final JdbcTypeDescription jdbcTypeDescription) {
         case Types.DECIMAL:
         case Types.NUMERIC:
             return mapNumericType(jdbcTypeDescription);
+        case Types.TIMESTAMP:
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleColumnMetadataReader.java]: This should be done by the base mapper.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
     // If set to true, the selectlist elements will get aliases such as c1, c2, ...
     // Can be refactored if we find a better way to implement it
     private boolean requiresSelectListAliasesForLimit = false;
+    private static final String TIMESTAMP_FORMAT = "'HH24:MI:SS.FF3 DD-MM-YYYY'";
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java]: English and US notation for date is usually `YYYY-MM-DD`.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java [hunk 3]
```diff
@@ -263,10 +264,13 @@ private String getProjectionString(final SqlColumn column, final String projecti
         final AbstractSqlDialect dialect = (AbstractSqlDialect) getDialect();
         final String typeName = ColumnAdapterNotes
                 .deserialize(column.getMetadata().getAdapterNotes(), column.getMetadata().getName()).getTypeName();
-        if ((typeName.startsWith("TIMESTAMP") && (((OracleSqlDialect) dialect).getImportType() == ImportType.JDBC))
-                || typeName.startsWith("INTERVAL") || typeName.equals("BINARY_FLOAT")
-                || typeName.equals("BINARY_DOUBLE") || typeName.equals("CLOB") || typeName.equals("NCLOB")) {
+        if (typeName.startsWith("INTERVAL") || typeName.equals("BINARY_FLOAT") || typeName.equals("BINARY_DOUBLE")
+                || typeName.equals("CLOB") || typeName.equals("NCLOB")) {
             return "TO_CHAR(" + projectionString + ")";
+        } else if (typeName.startsWith("TIMESTAMP")
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java]: Using `startsWith` has the potential to cause trouble later in the event of a newly introduced (and incompatible) timestamp format on the Oracle side.

It is safer to check for the explicit names.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 4]
```diff
@@ -671,11 +671,11 @@ void testDate() throws SQLException {
     void testTimestamp3() throws SQLException {
         final String query = "SELECT C11 FROM %s";
         final ResultSet resultJDBC = runQueryJDBC(query);
-        matchNextRow(resultJDBC, "11-MAR-13 05.30.15.123 PM");
+        matchNextRow(resultJDBC, Timestamp.valueOf("2013-03-11 17:30:15.123"));
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Reduce code duplication by extracting common part into method.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 5]
```diff
@@ -671,11 +671,11 @@ void testDate() throws SQLException {
...
         final ResultSet resultORA = runQueryORA(query);
         matchNextRow(resultORA, Timestamp.valueOf("2013-03-11 17:30:15.123"));
 
-        runMatchSingleRowExplainJDBC(query, "SELECT TO_CHAR(" + TYPE_TEST_T + ".\"C11\") FROM \"" + ORA_TABLE + "\"");
+        runMatchSingleRowExplainJDBC(query, "SELECT TO_TIMESTAMP(TO_CHAR(" + TYPE_TEST_T + ".\"C11\", 'HH24:MI:SS.FF3 DD-MM-YYYY'), 'HH24:MI:SS.FF3 DD-MM-YYYY') FROM \"" + ORA_TABLE + "\"");
         runMatchSingleRowExplainORA(query, "SELECT " + TYPE_TEST_T + ".\"C11\" FROM \"" + ORA_TABLE + "\"");
         assertEquals("TIMESTAMP", getColumnTypeJDBC("C11"));
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Use `assertThat`. Wrap with `assertAll`.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 6]
```diff
@@ -775,9 +775,9 @@ void testSelectAllTimestampColumns() throws SQLException {
 
         assertEquals("TIMESTAMP", columnTypesJDBC.get("A"));
```
Comment (redcatbear) [inline jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: Wrap with `assertAll`.

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleColumnMetadataReader.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleColumnMetadataReader.java]: You are right

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java]: Changed the format to: YYYY-MM-DD HH:MI:SS.FF6

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java]: Changed to type numbers

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 6 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java (hunk 5 above)
Comment (AnastasiiaSergienko) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java]: done

File: jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java (hunk 3 above)
Comment (redcatbear) [reply jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlGenerationVisitor.java]: Okay.

**Pull request merged**